cfme/utils/template/ec2.py

```diff
@@ -49,25 +49,47 @@ def upload_image(self):
 
     @log_wrap("import image from bucket")
     def import_image(self):
-        try:
-            import_task_id = self.mgmt.import_image(s3bucket=self.bucket_name,
-                                                    s3key=self.template_name,
-                                                    description=self.template_name)
-
-            wait_for(self.mgmt.get_image_id_if_import_completed,
-                     func_args=[import_task_id],
-                     fail_condition=False,
-                     delay=5,
-                     timeout='90m',
-                     message='Importing image to EC2')
-
-            ami_id = self.mgmt.get_image_id_if_import_completed(import_task_id)
-            ami = EC2Image(uuid=ami_id, system=self.mgmt)
-            ami.set_tag("Name", self.template_name)
-            return True
+        if self.stream == 'downstream-510z':
+            try:
+                import_task_id = self.mgmt.import_image(s3bucket=self.bucket_name,
+                                                        s3key=self.template_name,
+                                                        description=self.template_name)
+
+                wait_for(self.mgmt.get_image_id_if_import_completed,
+                         func_args=[import_task_id],
+                         fail_condition=False,
+                         delay=5,
+                         timeout='90m',
+                         message='Importing image to EC2')
+
+                ami_id = self.mgmt.get_image_id_if_import_completed(import_task_id)
+                ami = EC2Image(uuid=ami_id, system=self.mgmt)
+                ami.set_tag("Name", self.template_name)
+                return True
 
-        except Exception:
-            return False
+            except Exception:
+                return False
+        else:
+            try:
+                import_task_id = self.mgmt.import_snapshot(s3bucket=self.bucket_name,
+                                                        s3key=self.template_name,
+                                                        description=self.template_name)
+                wait_for(self.mgmt.get_snapshot_id_if_import_completed,
+                         func_args=[import_task_id],
+                         fail_condition=False,
+                         delay=5,
+                         timeout='90m',
+                         message='Importing image to EC2')
+
+                snapshot_id = self.mgmt.get_snapshot_id_if_import_completed(import_task_id)
+                ami_result = self.mgmt.create_image_from_snapshot(self.template_name, snapshot_id)
+                ami_id = ami_result.get("ImageId")
+                ami = EC2Image(uuid=ami_id, system=self.mgmt)
+                ami.set_tag("Name", self.template_name)
+                return True
+
+            except Exception:
```

Comment:
**required:** @mmojzis Please remove the `except Exception` from here and above or please do explain why you won't let the exception propagate. Letting it to propagate will lead to easier debugging. It is rarely a good idea to muffle exceptions. And even if we want to muffle some, we should specify which ones and not just simply catch all `Exception` instances.

https://realpython.com/the-most-diabolical-python-antipattern/

Reply:
Good point @jaryn, this is propagating a poor pattern that was introduced before this PR.

Given the use and scope of these template upload scripts, we do want to diaper exceptions here - but not without logging them.

Addition of `logger.exception` call before returning would be sufficient to log what occurred while maintaining some of the logging done by the wrapper/decorator for these methods.